apps/validator/src/main/scala/org/lfdecentralizedtrust/splice/validator/store/db/DbValidatorInternalStore.scala

```diff
@@ -4,73 +4,139 @@
+    version = 2,
+    name = "DbValidatorInternalConfigStore",
+    party = key.validatorParty,
+    participant = participant,
+    key = Map(
+      "validatorParty" -> key.validatorParty.toProtoPrimitive
+    ),
+  )
+
+  private case class ValidatorInternalStoreState(
+      storeId: Option[Int]
+  )
+
+  private object ValidatorInternalStoreState {
+    def empty(): ValidatorInternalStoreState = ValidatorInternalStoreState(
+      storeId = None
+    )
+  }
+
+  private val internalState =
+    new AtomicReference[ValidatorInternalStoreState](ValidatorInternalStoreState.empty())
+
+  def storeId: Int =
+    internalState
+      .get()
+      .storeId
+      .getOrElse(throw new RuntimeException("Using storeId before it was assigned"))
+
+  def initializeState()(implicit tc: TraceContext): Future[Unit] = {
+    val initializedResult: Future[Int] = StoreDescriptorStore
+      .getStoreIdForDescriptor(storeDescriptor, storage)
+    for {
+      result <- initializedResult
+    } yield {
+      internalState.updateAndGet(_.copy(storeId = Some(result)))
+      ()
+    }
+  }
 
   override def setConfig[T](key: String, value: T)(implicit
       tc: TraceContext,
       encoder: Encoder[T],
   ): Future[Unit] = {
-    val json = value.asJson
-    val action = sql"""INSERT INTO validator_internal_config (config_key, config_value)
-        VALUES ($key, $json)
-        ON CONFLICT (config_key) DO UPDATE
+    val jsonValue: Json = value.asJson
+
+    val action = sql"""INSERT INTO validator_internal_config (config_key, config_value, store_id)
+        VALUES ($key, $jsonValue, $storeId)
+        ON CONFLICT (store_id, config_key) DO UPDATE
         SET config_value = excluded.config_value""".asUpdate
+
     val updateAction = storage.update(action, "set-validator-internal-config")
+
     logger.debug(
-      s"saving validator config in database with key '$key' and value '$json'"
+      s"saving validator config in database with key '$key' and value '$jsonValue'"
     )
     updateAction.map(_ => ())
   }
 
-  override def getConfig[T](
+  def getConfig[T](
       key: String
   )(implicit tc: TraceContext, decoder: Decoder[T]): OptionT[Future, Decoder.Result[T]] = {
-    optionTfutureUnlessShutdownToFuture(
-      storage
-        .querySingle(
-          sql"""SELECT config_value
-        FROM validator_internal_config
-        WHERE config_key = $key
-      """.as[Json].headOption,
-          "get-validator-internal-config",
-        )
-    )
-      .map(_.as[T])
+
+    logger.debug(s"Retrieving config key $key")
+
+    val queryAction = sql"""SELECT config_value
+      FROM validator_internal_config
+      WHERE config_key = $key AND store_id = $storeId
+    """.as[Json].headOption
+
+    val jsonOptionT: OptionT[FutureUnlessShutdown, Json] =
+      storage.querySingle(queryAction, "get-validator-internal-config")
+
+    val resultOptionT: OptionT[FutureUnlessShutdown, Decoder.Result[T]] =
+      jsonOptionT.map { json =>
```

Comment:
```scala
for {
  json <- storage.querySingle(queryAction, "get-validator-internal-config")
} yield {
  val decodeResult = json.as[T]
  // ...
}
```
if you want to avoid some of the vals.